packages/pytest-simcore/src/pytest_simcore/rabbit_service.py

```diff
@@ -18,11 +18,27 @@
 
 from .helpers.utils_docker import get_service_published_port
 
+# HELPERS ------------------------------------------------------------------------------------
+
 log = logging.getLogger(__name__)
 
 
+@tenacity.retry(
+    wait=wait_fixed(5),
+    stop=stop_after_attempt(60),
+    before_sleep=before_sleep_log(log, logging.INFO),
+    reraise=True,
+)
+async def wait_till_rabbit_responsive(url: str) -> None:
+    connection = await aio_pika.connect(url)
+    await connection.close()
+
+
+# FIXTURES ------------------------------------------------------------------------------------
+
+
 @pytest.fixture(scope="module")
-def loop(request) -> asyncio.AbstractEventLoop:
+def loop(request) -> Iterator[asyncio.AbstractEventLoop]:

```

Comment:
I am actually confused whether the return type should be one of Iterable, Iterator, Generator....
see this [https://github.com/pytest-dev/pytest/discussions/7809](https://github.com/pytest-dev/pytest/discussions/7809)

Reply:
``Iterable`` or ``Iterator`` are specific cases of ``Generator[YieldType, SendType, ReturnType]`` but that does not have send or return  , i.e. ``Iterator = Generator[YieldType, None, None]``.  

See more details and examples in https://docs.python.org/3/library/typing.html#typing.Generator

Functionalitywise, ``Iterable`` and ``Iterator`` are equivalent and the difference is a bit more subtle. An iterator is iterable but not every iterable is an iterator (there are othr ways for python to deduce how to iterate). Taking from [this reference ](https://www.geeksforgeeks.org/python-difference-iterable-iterator):

> Iterable is an object, which one can iterate over. It generates an Iterator when passed to iter() method. Iterator is an > object, which is used to iterate over an iterable object using __next__() method. Iterators have __next__() method, > which returns the next item of the object.